spacetemplate/repository.go

```diff
@@ -0,0 +1,286 @@
+		}
+		return errs.Wrap(err, "failed to create space template")
+	}
+
+	// Apply the space template to a temporary space to see if it works and then
+	// delete everything
+	//defer cleaner.DeleteCreatedEntitiesNoTransaction(r.db)()
+
+	tempSpaceID := uuid.NewV4()
+	now := time.Now()
+	tempSpace := space.Space{
+		// this space is marked as being soft-deleted at creation time in order
+		// to not appear in any list
+		Lifecycle: gormsupport.Lifecycle{
+			DeletedAt: &now,
+		},
+		ID:              tempSpaceID,
+		Name:            tempSpaceID.String(),
+		Description:     fmt.Sprintf("This is a temporary space to test space template '%s'", st.ID),
+		SpaceTemplateID: st.ID,
+	}
+	spaceRepo := space.NewRepository(r.db)
+	sp, err := spaceRepo.Create(ctx, &tempSpace)
+	if err != nil {
+		log.Info(ctx, map[string]interface{}{
+			"space_id": tempSpaceID,
+			"err":      err,
+		}, "failed to create space '%s'")
+		return errs.Wrap(err, "failed to create space")
+	}
+	spaceToDelete := space.Space{ID: sp.ID}
+	defer r.db.Unscoped().Delete(spaceToDelete)
+	if err = r.apply(ctx, st.ID, tempSpaceID, true); err != nil {
+		log.Info(ctx, map[string]interface{}{
+			"space_template_id": st.ID,
+			"space_id":          tempSpaceID,
+			"err":               err,
+		}, "failed to apply space template to space")
+		return errs.Wrapf(err, "failed to apply space template '%s' to space '%s'", st.ID, tempSpaceID)
+	}
+
+	log.Info(ctx, map[string]interface{}{"space_template_id": st.ID}, "space template created successfully")
+	return nil
+}
+
+// List returns an array with all space templates in it
+func (r *GormRepository) List(ctx context.Context) ([]SpaceTemplate, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "space_template", "list"}, time.Now())
+	var objs []SpaceTemplate
+	err := r.db.Find(&objs).Error
+	if err != nil && err != gorm.ErrRecordNotFound {
+		return nil, errs.Wrap(err, "failed to list space templates")
+	}
+	return objs, nil
+}
+
+// Load returns a single space template by a given ID
+func (r *GormRepository) Load(ctx context.Context, spaceTemplateID uuid.UUID) (*SpaceTemplate, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "space_template", "load"}, time.Now())
+	var obj SpaceTemplate
+	tx := r.db.Where("id = ?", spaceTemplateID).First(&obj)
+	if tx.RecordNotFound() {
+		log.Info(ctx, map[string]interface{}{
+			"space_template_id": spaceTemplateID.String(),
+		}, "space template not found")
+		return nil, errors.NewNotFoundError("space_template", spaceTemplateID.String())
+	}
+	if tx.Error != nil {
+		return nil, errors.NewInternalError(tx.Error.Error())
+	}
+	return &obj, nil
+}
+
+func (r *GormRepository) Apply(ctx context.Context, spaceTemplateID uuid.UUID, spaceID uuid.UUID) error {
+	return r.apply(ctx, spaceTemplateID, spaceID, false)
+}
+
+func (r *GormRepository) apply(ctx context.Context, spaceTemplateID uuid.UUID, spaceID uuid.UUID, deleteArtifacts bool) error {
```

Comment:
why do you need this `deleteArtifacts` arg in the end ? when you try to create the space template, you're going to create a 'test' space in which all artifacts will be created, and then upon completion, the space will be delete. Shouldn't all space artifacts be deleted at the same time (ie, in cascade) ?